video/out/vo_vaapi_wayland.c

```diff
@@ -0,0 +1,371 @@
+			return VO_ERROR;
+		p->solid_buffer_pool = wl_shm_create_pool(wl->shm, fd, height * stride);
+		if (!p->solid_buffer_pool)
+			return VO_ERROR;
+		p->solid_buffer = wl_shm_pool_create_buffer(p->solid_buffer_pool, 0,
+				width, height, stride, WL_SHM_FORMAT_XRGB8888);
+		if (!p->solid_buffer)
+			return VO_ERROR;
+	}
+	if (!vo_wayland_reconfig(vo))
+		return VO_ERROR;
+
+	return 0;
+}
+
+static int resize(struct vo *vo) {
+	struct vo_wayland_state *wl = vo->wl;
+	struct priv *p = vo->priv;
+
+	wl_subsurface_set_sync(wl->video_subsurface);
+	vo_wayland_set_opaque_region(wl, 0);
+	const int32_t width = wl->scaling * mp_rect_w(wl->geometry);
+	const int32_t height = wl->scaling * mp_rect_h(wl->geometry);
+	vo->dwidth = width;
+	vo->dheight = height;
+	vo_get_src_dst_rects(vo, &p->src, &p->dst, &p->osd);
+	wp_viewport_set_destination(wl->viewport,
+			(p->dst.x0 << 1) + mp_rect_w(p->dst),
+			(p->dst.y0 << 1) + mp_rect_h(p->dst));
+	wp_viewport_set_destination(wl->video_viewport, mp_rect_w(p->dst),
+			mp_rect_h(p->dst));
+	wl_subsurface_set_position(wl->video_subsurface, p->dst.x0, p->dst.y0);
+	vo->want_redraw = true;
+	wl_subsurface_set_desync(wl->video_subsurface);
+
+	return VO_TRUE;
+}
+
+static int control(struct vo *vo, uint32_t request, void *data) {
+	int events = 0;
+	int ret = vo_wayland_control(vo, &events, request, data);
+
+	if (events & VO_EVENT_RESIZE)
+		ret = resize(vo);
+	if (events & VO_EVENT_EXPOSE)
+		vo->want_redraw = true;
+	vo_event(vo, events);
+
+	return ret;
+}
+
+static void draw_frame(struct vo *vo, struct vo_frame *frame) {
+	struct priv *p = vo->priv;
+	struct vo_wayland_state *wl = vo->wl;
+
+	if (!vo_wayland_check_visible(vo))
+		return;
+
+	struct va_pool_entry *entry = va_pool_alloc_entry(p->va_pool,
+			frame->current);
+	if (!entry)
+		return;
+
+	wl_surface_attach(wl->surface, p->solid_buffer, 0, 0);
+	wl_surface_attach(wl->video_surface, entry->buffer, 0, 0);
+	wl_surface_damage_buffer(wl->video_surface, 0, 0, INT32_MAX, INT32_MAX);
+	wl_surface_commit(wl->video_surface);
+	wl_surface_commit(wl->surface);
+
+	if (!wl->opts->disable_vsync)
+		vo_wayland_wait_frame(wl);
+	if (wl->presentation)
+		vo_wayland_sync_swap(wl);
+}
+static void flip_page(struct vo *vo) {
+	/* no-op */
+}
+
+static void get_vsync(struct vo *vo, struct vo_vsync_info *info) {
+	struct vo_wayland_state *wl = vo->wl;
+	if (wl->presentation) {
+		info->vsync_duration = wl->vsync_duration;
+		info->skipped_vsyncs = wl->last_skipped_vsyncs;
+		info->last_queue_display_time = wl->last_queue_display_time;
+	}
+}
+
+const struct vo_driver video_out_vaapi_wayland = { .description =
+		"VA API with Wayland video output", .name = "vaapi-wayland", .preinit =
+		preinit, .query_format = query_format, .reconfig = reconfig, .control =
+		control, .draw_frame = draw_frame, .flip_page = flip_page, .get_vsync =
+		get_vsync, .wakeup = vo_wayland_wakeup, .wait_events =
+		vo_wayland_wait_events, .uninit = uninit, .priv_size =
+		sizeof(struct priv), };
```

Comment:
Could this struct be more organized so it's just one thing per line like how `vo_wlshm` does it?